src/libraries/Microsoft.PowerFx.Core/Binding/ReadOnlyINameResolver.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core
+{
+    /// <summary>
+    /// Helper class to create composable INameResolver. Only supports LookupType method currently.
+    /// Other INameResolver methods should be implemented as needed.
+    /// </summary>
+    internal class ReadOnlyINameResolver : ReadOnlySymbolTable, INameResolver
```

Comment:
>ReadOnlyINameResolver [](http://example.com/codeflow?start=19&length=21)

Suggestion: `ReadOnlyNameResolver` (no need for `I` prefix in the middle of the name)